src/frontends/tensorflow_common/src/op/adjust_saturation.cpp

```diff
@@ -0,0 +1,202 @@
+
+    // r == vv: hh = norm * (g - b)
+    auto hue_case_r = make_shared<v1::Multiply>(norm, make_shared<v1::Subtract>(g, b));
+
+    // g == vv: hh = norm * (b - r) + 2.0 / 6.0
+    auto const_2_by_6 = make_shared<v0::Constant>(type, Shape{}, 2.0f / 6.0f);
+    auto hue_case_g = make_shared<v1::Add>(
+        make_shared<v1::Multiply>(norm, make_shared<v1::Subtract>(b, r)),
+        const_2_by_6
+    );
+
+    // b == vv: hh = norm * (r - g) + 4.0 / 6.0
+    auto const_4_by_6 = make_shared<v0::Constant>(type, Shape{}, 4.0f / 6.0f);
+    auto hue_case_b = make_shared<v1::Add>(
+        make_shared<v1::Multiply>(norm, make_shared<v1::Subtract>(r, g)),
+        const_4_by_6
+    );
+
+    // select hue based on the maximum component
+    // check if `r` is the max, otherwise check if `g` is the max, if not use `b`'s hue
+    auto hh = make_shared<v1::Select>(
+        r_eq_v,
+        hue_case_r,  // Use hue_case_r if r is max
+        make_shared<v1::Select>(
+            g_eq_v,
+            hue_case_g,  // Use hue_case_g if g is max
+            hue_case_b   // Use hue_case_b otherwise (b is max)
+        )
+    );
+
+    // range = 0.0: hh = 0
+    auto hh_zero_range = make_shared<v1::Select>(make_shared<v1::Equal>(range, const_zero_f_), const_zero_f_, hh);
+
+    // hh < 0.0: hh = hh + 1
+    auto hh_final = make_shared<v1::Select>(make_shared<v1::Less>(hh, const_zero_f_), make_shared<v1::Add>(hh_zero_range, const_one_f_), hh_zero_range);
+
+    return make_shared<tuple<shared_ptr<Node>, shared_ptr<Node>, shared_ptr<Node>>>(hh_final, ss, vv);
+}
+
+shared_ptr<Node> hsv_to_rgb(shared_ptr<Node> h, shared_ptr<Node> s, shared_ptr<Node> v, element::Type type) {
+    // image format conversion based on https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/kernels/image/adjust_saturation_op.cc
+    auto const_six_f_ = make_shared<v0::Constant>(type, Shape{}, 6.0f);
+    auto const_two_f_ = make_shared<v0::Constant>(type, Shape{}, 2.0f);
+    auto const_one_f_ = make_shared<v0::Constant>(type, Shape{}, 1.0f);
+
+    auto const_minus_one_i_ = make_shared<v0::Constant>(element::i32, Shape{}, -1);
+    auto const_minus_two_i_ = make_shared<v0::Constant>(element::i32, Shape{}, -2);
+    
+    // c = s * v;
+    auto c = make_shared<v1::Multiply>(s, v);
+    // m = v - c;
+    auto m = make_shared<v1::Subtract>(v, c);
+    // dh = h * 6;
+    auto dh = make_shared<v1::Multiply>(h, const_six_f_);
+
+    // fmodu rounded to within [0, 2)
+    auto fmodu = make_shared<v1::FloorMod>(dh, const_two_f_);
+
+    //  x = c * (1 - std::abs(fmodu - 1));
+    auto x = make_shared<v1::Multiply>(c, make_shared<v1::Subtract>(const_one_f_, make_shared<v0::Abs>(make_shared<v1::Subtract>(fmodu, const_one_f_))));
+
+    // h_category: [batch_dims..., H, W, 1]
+    auto h_category = make_shared<v0::Convert>(make_shared<v0::Floor>(dh), element::i32);
+
+    auto zeros = make_shared<v0::Constant>(type, x->get_shape(), 0.0f);
```

Comment:
please don't use `get_shape`. It can raise an exception for partial shapes that is not rare case.
Instead, I suggest to create a scalar zero const and broadcast it to x's shape: x->ShapeOf->Broadcast(const_zero) sub-graph